cookbooks/cdo-nodejs/recipes/default.rb

```diff
@@ -3,12 +3,36 @@
 # Recipe:: default
 #
 
-node.default['nodejs']['repo'] = "https://deb.nodesource.com/node_#{node['cdo-nodejs']['version']}"
-include_recipe 'nodejs'
+is_ubuntu_18_04 = node["platform"] == "ubuntu" && node["platform_version"] == "18.04"
+is_node_18 = node['cdo-nodejs']['version'] == '18.x'
 
-# Keep nodejs up to date
-package 'nodejs' do
-  action :upgrade
+if is_ubuntu_18_04 && is_node_18
+  # No nodesource binary for NodeJS 18 on Ubuntu 18.04
+  # Compile from source instead, using gcc-8
+
+  build_essential 'install build tools'
+
+  package %w(libssl-dev python g++ g++-8 gcc-8)
+
+  ark "nodejs-source" do
+    url "https://nodejs.org/dist/v18.16.0/node-v18.16.0.tar.gz"
+    version "18.16.0"
+    make_opts ["-j 8"]
```

Comment:
I had it coded that way originally, but backed off it, because:
1) The context this will be used in are adhocs, where we want to use ALL the CPU (currently 8, hence the -j 8), and prod/staging where we probably DO NOT want to use all the CPUs, and 8 of 48 felt like a reasonable fraction to use while chef-ing live.
2) The main place where `-j 8` might not be a good value is probably in a local dev kitchen dokker env, but I tested and having -j too high isn't a big negative impact, and if you're gonna have an impact, it feels better to do it on local dev.
3) With that much complexity, the shell got relatively unwieldy and ruined simplicity and clarity for not much gain

Reply:
Hah, yeah, I hadn't considered that what we might actually want here is "the number of processors available but only up to a limit because we sometimes have an unreasonable amount". That makes sense to me! I'd love to see a comment summarizing those reasons

Reply:
I kinda think this is over-commenting, maybe enough that the PR is linked to this change and has the discuss here? Not gonna fight ya if you got a strong intuition future devs will want to see this 😌

Reply:
Nah, no strong intuition. I mention it just because these chef scripts tend to go stale quickly, and we don't have a lot of institutional familiarity with these scripts or with chef in general, so I err way on the side of over-commenting. You're right that it probably is unnecessary but that ain't gonna stop me from at least suggesting it :smile: